Android/src/org/droidplanner/android/proxy/mission/item/fragments/MissionDetailFragment.java

```diff
@@ -206,11 +208,69 @@ public void onSpinnerItemSelected(Spinner spinner, int position) {
         }
     };
 
+    private final Spinner.OnItemSelectedListener missionFrameListener = new Spinner.OnItemSelectedListener() {
+
+        @Override
+        public void onItemSelected(AdapterView<?> parent, View view, int position, long id) {
+            frameSpinner = (Spinner) view.findViewById(R.id.frameSpinner);
+
+            if (frameSpinner != null && mSelectedProxies.isEmpty())
+                return;
+
+            MissionItemType selectedType = commandAdapter.getItem(position);
+            List<Pair<MissionItemProxy, List<MissionItemProxy>>> updatedList = new ArrayList<>(
+                    mSelectedProxies.size());
+
+            for (MissionItemProxy missionItemProxy : mSelectedProxies) {
+
+                MissionItem currentItem = missionItemProxy.getMissionItem();
+
+                List<MissionItemProxy> updatedItems = new ArrayList<>();
+
+                if (currentItem instanceof MissionItem.SpatialItem) {
+                    // Only items that have frames need to be updated
+                    MissionItem.SpatialItem spatialItem = ((MissionItem.SpatialItem) currentItem);
+
+                    boolean updated = false;
+                    switch (position) {
+                        case 0:
+                            updated = spatialItem.getCoordinate().setFrame(Frame.GLOBAL_ABS);
+                            break;
+                        case 1:
+                            updated = spatialItem.getCoordinate().setFrame(Frame.GLOBAL_RELATIVE);
+                            break;
+                        case 2:
+                            updated = spatialItem.getCoordinate().setFrame(Frame.GLOBAL_TERRAIN);
+                            break;
+                        default:
+                            // Do Nothing
+                            break;
+                    }
+
+                    if (updated) {
+                        updatedItems.add(missionItemProxy);
+                        updatedList.add(Pair.create(missionItemProxy, updatedItems));
+                    }
+                }
+            }
+            if (!updatedList.isEmpty()) {
+                mListener.onWaypointTypeChanged(selectedType, updatedList);
+                dismiss(); // TODO:!BB! do we want to dismiss the dialog
```

Comment:
@billbonney I don't think so. It should auto dismiss after the user's selection.